src/features/todos.ts

```diff
@@ -1,8 +1,41 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
 import { Todo } from '../types/Todo';
+import { getTodos } from '../api';
+
+type InitialState = {
+  todos: Todo[];
+  loaded: boolean;
+  selectedTodo: Todo | null;
+};
+
+const initialState: InitialState = {
+  todos: [],
+  loaded: false,
+  selectedTodo: null,
+};
+
+export const init = createAsyncThunk('todos/fetch', () => {
+  return getTodos();
+});
 
 export const todosSlice = createSlice({
   name: 'todos',
-  initialState: [] as Todo[],
-  reducers: {},
+  initialState,
+  reducers: {
+    setSelectedTodo: (state, action: PayloadAction<Todo | null>) => {
+      return { ...state, selectedTodo: action.payload };
+    },
+  },
+  extraReducers(builder) {
+    builder.addCase(init.pending, state => {
+      return { ...state, loaded: true };
```

Comment:
The `loaded` state should be set to `false` in the `init.pending` case to indicate that data is being fetched. This will ensure that the `Loader` component is displayed during data fetching.